devicetypes/smartthings/zigbee-sound-sensor/zigbee-sound-sensor.groovy

```diff
@@ -0,0 +1,173 @@
+        log.debug "enroll response: ${cmds}"
+        result = cmds?.collect { new physicalgraph.device.HubAction(it)}
+    }
+    return result
+}
+
+private Map parseIasMessage(String description) {
+    ZoneStatus zs = zigbee.parseZoneStatus(description)
+    def result = [:]
+    if(zs.isAlarm1Set() || zs.isAlarm2Set()) {
+        result = getSoundDetectionResult("detected")
+    } else if(!zs.isTamperSet()) {
+        result = getSoundDetectionResult("not detected")
+    } else {
+        result = [displayed: true, descriptionText: "${device.displayName}'s case is opened"]
+    }
+
+    return result
+}
+
+private Map parseAttrMessage(description) {
+    def descMap = zigbee.parseDescriptionAsMap(description)
+    def map = [:]
+    if(descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+        map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+    } else if(descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+        if (descMap.data[0] == "00") {
+            sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+        } else {
+            log.warn "TEMP REPORTING CONFIG FAILED - error code: ${descMap.data[0]}"
+        }
+    } else if(descMap.clusterInt == 0x0020 && descMap.commandInt == 0x00) {
+        log.debug "Check in interval command received!"
+        sendEvent(name: "checkInterval", value: 60 * 60, displayed: true, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    } else if (map.name == "temperature") {
+        if (tempOffset) {
+            map.value = (int) map.value + (int) tempOffset
+        }
+        map.descriptionText = temperatureScale == 'C' ? '${device.displayName} was ${value}°C' : '${device.displayName} was ${value}°F'
+        map.translatable = true
+    }
+
+    return map
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+    def result = [:]
+    def volts = rawValue / 10
+    if (!(rawValue == 0 || rawValue == 255)) {
+        def minVolts = 2.2
+        def maxVolts = 3.0
+        def pct = (volts - minVolts) / (maxVolts - minVolts)
+        def roundedPct = Math.round(pct * 100)
+        if (roundedPct <= 0)
+            roundedPct = 1
+        result.value = Math.min(100, roundedPct)
+    }
+    log.debug "Battery Percentage rawValue = ${rawValue} -> ${(rawValue-23)/7.0 * 100}"
+    result.name = 'battery'
+    result.translatable = true
+    result.descriptionText = "${device.displayName} battery was ${result.value}%"
+    return result
+}
+
+private Map getSoundDetectionResult(value) {
+    def text = "Sound was ${value}"
+    def result = [name: "sound", value: value, descriptionText: text, displayed: true]
+    return result
+}
+
+def ping() {
+    log.debug "Ping call"
+    return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
```

Comment:
I don't think the debug messages are important enough to justify the code duplication between here and `refresh`